src/Tracker.cc

```diff
@@ -1853,3 +1912,176 @@ void Tracker::CollectNearestPoints(KeyFrame& kf, std::set<MapPoint*>& spNearestP
   }
 }
 
+void Tracker::UpdateCamsFromWorld(MultiKeyFrame* mpTempMKF)
+{
+  for(KeyFramePtrMap::iterator kf_it = mpTempMKF->mmpKeyFrames.begin(); kf_it != mpTempMKF->mmpKeyFrames.end(); ++kf_it)
+  {
+    kf_it->second->mse3CamFromWorld = kf_it->second->mse3CamFromBase * mpTempMKF->mse3BaseFromWorld; // CHECK!! GOOD
+  }
+}
+
+void Tracker::RecordMeasurementsAndBufferKeyFrame() 
+{
+       
+  //save all the tracker measurements
+    
+  //create a temporary MKF, copy over current MKF
+  MultiKeyFrame* mpTempMKF =  mpCurrentMKF->CopyMultiKeyFramePartial();
+  UpdateCamsFromWorld(mpTempMKF);
+  mpTempMKF->isBufferMKF = true;
```

Comment:
Why do you need the copy, why not work on the CurrentMKF and only store it if its a likely (the current best) candidate?